hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/CryptoCreateValidator.java

```diff
@@ -86,19 +86,24 @@ public void validateKey(
         }
     }
 
-    /** Check if the number of auto associations is too many.
+    /** Check if the number of auto associations is too many
+     * or in the case of unlimited auto associations, check if the number is less than -1 or 0 if disabled.
      * @param n number to check
      * @param ledgerConfig LedgerConfig
      * @param entitiesConfig EntitiesConfig
      * @param tokensConfig TokensConfig
      * @return true the given number is greater than the max number of auto associations
+     * or negative and unlimited auto associations are disabled
+     * or less than -1 if unlimited auto associations are enabled
      */
     public boolean tooManyAutoAssociations(
             final int n,
             @NonNull final LedgerConfig ledgerConfig,
             @NonNull final EntitiesConfig entitiesConfig,
             @NonNull final TokensConfig tokensConfig) {
-        return n > ledgerConfig.maxAutoAssociations()
-                || (entitiesConfig.limitTokenAssociations() && n > tokensConfig.maxPerAccount());
+        return (entitiesConfig.limitTokenAssociations() && n > tokensConfig.maxPerAccount())
+                || n > ledgerConfig.maxAutoAssociations()
+                || (n < -1 && entitiesConfig.unlimitedAutoAssociationsEnabled())
```

Comment:
n  < -1 is not allowed right ? We should throw in pureChecks

Reply:
Yes, exactly. This method checks if the auto associations are out of limits.